nemo_skills/evaluation/metrics/ioi_metrics.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from nemo_skills.evaluation.metrics.base import BaseMetrics
+from collections import defaultdict
+
+
+class IOIMetrics(BaseMetrics):
+    def __init__(self):
+        super().__init__()
+        self.reset()
+
+    def update(self, predictions):
+        if len(predictions) > 1:
```

Comment:
should it be better to call super().update(predictions) and then only add your changes? Also agg_mode greedy is no longer supported by any other benchmark, should use pass@1 instead

Reply:
I updated it as much as I can, let me know if this is what you had in mind